hw/top_earlgrey/data/chip_testplan.hjson

```diff
@@ -1041,32 +1207,133 @@
     }
 
     // OTP (pre-verified IP) integration tests:
-    // TODO: more testing needed on OTP.
     {
-      name: chip_otp_init
-      desc: '''Verify the OPT initialization on chip power up.
+      name: chip_otp_ctrl_init
+      desc: '''Verify the OTP ctrl initialization on chip power up.
 
-            Details TBD.
+            Verify that the chip powers up correctly on POR.
+            - The pwrmgr initiates a handshake with OTP ctrl and later, with LC ctrl in subsequent
+              FSM states. Ensure that the whole power up sequence does not hang.
+            - Verify with connectivity assertion checks, the handshake signals are connected.
+            - Ensure that no interrupts or alerts are triggered.
             '''
       milestone: V2
       tests: []
     }
     {
-      name: chip_otp_keys
+      name: chip_otp_ctrl_keys
       desc: '''Verify the proliferation of keys to security peripherals.
 
-            Ensure that the correct set of keys are provided to sram, sram_ret, flash, keymgr and
-            the OTBN. X-ref'ed with those individual IP tests.
+            - Verify the correctness of keys provided to SRAM ctrl (main & ret), flash ctrl, keymgr,
+              (note that keymgr does not have handshake), and OTBN.
+
+            X-ref'ed with IP tests that consume these signals.
             '''
       milestone: V2
       tests: []
     }
     {
-      name: chip_otp_lc_program
-      desc: '''Verify the OTP program req from LC.
+      name: chip_otp_ctrl_entropy
+      desc: '''Verify the entropy interface from OTP ctrl to EDN.
 
-            Ensure that the correct set of keys are provided to sram, sram_ret, flash, keymgr and
-            the OTBN. X-ref'ed with those individual IP tests.
+            This is X-ref'ed with the chip_otp_ctrl_keys test, which needs to handshake with the EDN
+            to receive some entropy bits before the keys for SRAM ctrl and OTBN are computed.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_otp_ctrl_edn_reset
+      desc: '''Verify the effect of putting the EDN domain in reset on OTP ctrl.
+
+            Verify that the computed nonce for SRAM is reset?
+            Verify all entropy data used for various things are reset.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_otp_ctrl_program
+      desc: '''Verify the program request from lc_ctrl.
```

Comment:
Another potential test is that when either lc or otp are busy any transition to low power is blocked. Incidentally, flash busy also blocks such transitions.

Reply:
Good point! This is something we should test as well.

Reply:
It will be covered in the pwrmgr testplan, which we will review once Tim is back.